### ISSUE TITLE
Statements after infinite loops are not considered unreachable

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#648]
The following example should not emit a diagnostic
```py
def f():
    while True:
        pass

    foo  # error: unresolved-reference
```
https://play.ty.dev/9108a9b2-63fd-42d7-8a3c-5810509cdd4a
